src/main/java/org/mule/tools/devkit/sonar/checks/java/NumberOfSimpleAndOptionalArgumentsCheck.java

```diff
@@ -0,0 +1,47 @@
+package org.mule.tools.devkit.sonar.checks.java;
+
+import com.google.common.collect.Iterables;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.mule.tools.devkit.sonar.ConnectorCertificationRulesDefinition;
+import org.mule.tools.devkit.sonar.utils.ClassParserUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.check.Rule;
+import org.sonar.check.RuleProperty;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.MethodTree;
+import org.sonar.squidbridge.annotations.ActivatedByDefault;
+
+@Rule(key = NumberOfSimpleAndOptionalArgumentsCheck.KEY, name = "Too many @Optional arguments in a processor", description = "Checks, for every @Processor, that the number of optional arguments (each of which must be of simple-type) doesn't exceed the maximum allowed. If it does, the suggested approach is to wrap them all in a separate POJO class. ", tags = { "connector-certification" })
+@ActivatedByDefault
+public class NumberOfSimpleAndOptionalArgumentsCheck extends AbstractConnectorClassCheck {
+
+    private static final Logger logger = LoggerFactory.getLogger(NumberOfSimpleAndOptionalArgumentsCheck.class);
+    public static final String KEY = "number-of-simple-and-optional-arguments";
+    private static final RuleKey RULE_KEY = RuleKey.of(ConnectorCertificationRulesDefinition.REPOSITORY_KEY, KEY);
+
+    @Override
+    protected RuleKey getRuleKey() {
+        return RULE_KEY;
+    }
+
+    private static final int DEFAULT_MAX_ALLOWED = 6;
+
+    /**
+     * The maximum number of simple-type arguments allowed in a processor.
+     */
+    @RuleProperty(key = "maxArgumentsAllowed", defaultValue = "" + DEFAULT_MAX_ALLOWED, description = "The maximum number of complex-type arguments allowed in a method annotated with @Processor")
```

Comment:
Please update description.
Remember that description field supports markdown, so feel free to play with those ;-)